packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -237,12 +302,26 @@ export function useConnections({
         return false;
       }
     },
-    [openToast, saveConnection]
+    [
+      openToast,
+      repositorySaveConnection,
+      findConnectionInfo,
+      getActiveAndInactiveCount,
+      connectionsManager,
+      onConnectionCreated,
+    ]
   );
 
   const removeConnection = useCallback(
     async (connectionInfo: ConnectionInfo) => {
-      await deleteConnection(connectionInfo);
+      await repositoryRemoveConnection(connectionInfo);
+      const { active, inactive } = getActiveAndInactiveCount();
```

Comment:
you mean it's stale when we get them here or by the time the track is send?

Reply:
for the first I thought having `getActiveAndInactiveCount` as a dependency would do the trick, doesn't it? for the latter I'm not sure if we anything fresher than the state after the removal

Reply:
When you get them here. Function itself and all the values inside of it are going to be potentially stale at the moment of calling because it is called after the async operation, it doesn't really matter whether or not it's a React "callback" and what dependencies you provide here, there is no way to safely access values like that if you're not reading them from the state directly.

This is a common issue with closures, just harder to spot due to all the extra code we have here, if you'd simplify it to the very core, it would look something like this, hopefully it highlights better what is going wrong here:

```
// simulate react render call: when component renderes, everything defined
// inside the scope will be part of it and will not change
function render(recentConnections) {
  const getRecentConnectionsLength = () => recentConnections.length;
  const removeRecent = () => {
    setTimeout(() => {
      // pretend we did some async work here
      console.log(getRecentConnectionsLength())
    }, 0);
  }
  return { removeRecent };
}
var recentConnections = [];
var { removeRecent } = render(recentConnections);
removeRecent(); // doesn't log immediately, async operation is in progress
// we changed the value and triggered a re-render, but it doesn't matter because the
// function that we already triggered still has the old value in the scope
recentConnections = [1, 2, 3];
void render(recentConnections);
// async op is done and you'll see zero logged
```

This is also very similar to the common issue of accessing a stale state in redux thunks and why they recommend to always call `getState()` inside of a thunk action instead of doing it once at the beginning of async flow.